src/components/PipelineRunDetailsView/PipelineRunDetailsView.tsx

```diff
@@ -38,12 +40,25 @@ export const PipelineRunDetailsView: React.FC<
   const [taskRuns, taskRunsLoaded, taskRunError] = useTaskRuns(namespace, pipelineRunName);
   const [canPatchComponent] = useAccessReviewForModel(ComponentModel, 'patch');
   const [canPatchPipeline] = useAccessReviewForModel(PipelineRunModel, 'patch');
+  const [canPatchSnapshot] = useAccessReviewForModel(SnapshotModel, 'patch');
 
   const [component, componentLoaded, componentError] = useComponent(
     namespace,
     pipelineRun?.metadata?.labels?.[PipelineRunLabel.COMPONENT],
   );
 
+  const [snapshots, snapshotsLoaded, snapshotsError] = useSnapshots(namespace);
+
+  const snapshot = React.useMemo(
+    () =>
+      snapshotsLoaded &&
+      !snapshotsError &&
+      snapshots.find(
+        (sn) => sn.metadata?.name === pipelineRun?.metadata?.labels?.[PipelineRunLabel.SNAPSHOT],
+      ),
+    [snapshots, snapshotsLoaded, snapshotsError, pipelineRun?.metadata?.labels],
+  );
```

Comment:
instead of using all the labels as dependencies, just use the value of that particular label.
i.e. pipelineRun?.metadata?.labels?.[PipelineRunLabel.SNAPSHOT]

```suggestion
const snapshotName = pipelineRun?.metadata?.labels?.[PipelineRunLabel.SNAPSHOT]
  const snapshot = React.useMemo(
    () =>
      snapshotsLoaded &&
      !snapshotsError &&
      snapshots.find(
        (sn) => sn.metadata?.name === snapshotName,
      ),
    [snapshots, snapshotsLoaded, snapshotsError, snapshotName],
  );
```

Reply:
lint error